### PR TITLE
Link in text block

### PR DESCRIPTION
**Pull request by WilcoFiers**
Changed
- Rule is renamed
- Broke out two algorithms
- Look at border-top / border-bottom / outline, instead of the generic 'border'
- Method for locating the background color has been refined
- Assumption added about border color

### CODE REVIEW

File: drafts/SC1-4-1-link-in-text-style.md [hunk 1]
```diff
@@ -1,127 +1,172 @@
...
 - [C15: Using CSS to change the presentation of a user interface component when it receives focus](http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140916/C15:)
 - [Color contrast (WCAG 2.0 definition)](http://www.w3.org/TR/WCAG20/#contrast-ratiodef)
 
-
 ## Assumptions
+
 - This test assumes that the 3:1 contrast difference between text is minimal to what would be sufficient to meet WCAG 2.0. This value is part of G183 technique, but is not specified in the 1.4.1 success criterion.
-- This tests assumes that one of the following shorthand CSS-properties (or the related expanded properties) is used to make the link visually evident: background, border, color, font, or text-decoration.
+- This tests assumes that one of the following shorthand CSS-properties (or the related expanded properties) is used to make the link visually evident: `background`, `border`, `color`, `font`, or `text-decoration`.
 - This test assumes that any use of border will make links sufficiently distinguishable
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: redundant to the last point: Use of a `border`, of 1 or more pixels, not set to none, and not with a color of transparent, is assumed to be distinguishable

File: drafts/SC1-4-1-link-in-text-style.md [hunk 2]
```diff
@@ -1,127 +1,172 @@
...
 | Test environment  | Remote Controlled User Agent
 | Test subject      | Single web page or DOM document fragment
 
 *Note*: Tools only able to process HTML + CSS can implement the first two step and ignore step 3 through 4, returning 'cantTell' instead. A tool could ask the user to perform step 3 and 4 manually, in which case the test would be semi-automatic instead.
 
-
 ## Test procedure
 
+*Note*: This rule requires applying focus and hover. The page should be restored to it's original test state after this rule hs completed.
+
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: typo: this rule (h)is completed

File: drafts/SC1-4-1-link-in-text-style.md [hunk 3]
```diff
@@ -1,127 +1,172 @@
...
-Select each element that matches `a[href]` that meets the following requirements:
-- link.textContent is [[non-empty]] text
-- Its nearest ancestor with its display style set to block has:
-** A different text color then the text color of the link
-** One or more child nodes of type text or containing text before or after the selected link
+Test mode: [automatic][AUTO]
 
-### Step 1
-Test method: [automatic]
+1. Select each element that matches `a[href]:not(role), *[role=link]`
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: couldn't it also be a link with role=link?

File: drafts/SC1-4-1-link-in-text-style.md [hunk 4]
```diff
@@ -1,127 +1,172 @@
...
+Test mode: [automatic][AUTO]
 
-Determine color and background-color of the link and the surrounding text and compute<br />C1 = contrast difference of surrounding text color and link color<br />C2 = contrast difference surrounding text background color and link background color
+Determine `color` and `background-color` of the link and it's block-like ancestor. If the background-color is transparent, use getElementsFromPoint to locate the closest underlying element that does have a `background-color` and use that value.
 
-If C1 or C2 is more than 3:1, continue to [[#Step 3]]
+1. C1 = contrast difference of surrounding text `color` and link `color`
 
-Else return
+2. C2 = contrast difference surrounding text `background-colo` and link `background-color`
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: typo: `background-colo`

File: drafts/SC1-4-1-link-in-text-style.md [hunk 5]
```diff
@@ -1,127 +1,172 @@
...
+- The `*-style` is not set to `none`
+- The `*-color` is not `transparent`
+
+If either `border-top`, `border-bottom` or `outline` is distinguishable, the link has a distinguishing border.
+
+### Distinguishing Style 
+
+Next, compare the style properties `font-family`, `font-weight`, `font-style`, `text-decoration` of the link, to that of the block-like ancestor. 
+
+If any of the properties have a different value for the one then for the other, the style is distinguishable.
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: typo: for the one tha(e)n for

File: drafts/SC1-4-1-link-in-text-style.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+| ID       | {{ page.name }}-fail3
+| Error    | The link is not sufficiently distinguishable from the surrounding text when it is hovered over with the mouse pointer.
+
+## Algorithms
+
+### Block-like element
+
+From the element, get the computed CSS value of the `display` property.
+
+Check that the value is not `inline` or starts with `inline-`.
```
Comment (rdeltour) [inline drafts/SC1-4-1-link-in-text-style.md]: Might be too simplistic, by looking at the set of [allowed values on the `display` property](https://developer.mozilla.org/en-US/docs/Web/CSS/display). For instance, an element with a  display value of `ruby` is considered inline (I had never heard of this display property before, TBH).
Needs extra review.

File: drafts/SC1-4-1-link-in-text-style.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+## Description
+
+This rule checks that links that are embedded in a block of text can be distinguished in a way that does not rely exclusively on color. If this is the case, the link can only pass if the color contrast is sufficient, and additional cues are given when the link receives focus or hover.
+
+## Background
+
+- [F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140311/F73.html)
+- [G183: Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](http://www.w3.org/TR/2013/NOTE-WCAG20-TECHS-20130905/G183)
+- [C15: Using CSS to change the presentation of a user interface component when it receives focus](http://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140916/C15:)
```
Comment (YetAnotherEye) [inline drafts/SC1-4-1-link-in-text-style.md]: http://www.w3.org/TR/2013/NOTE-WCAG20-TECHS-20130905/G183
should be 
https://www.w3.org/TR/2014/NOTE-WCAG20-TECHS-20140311/G183
(assuming that 2014 is still our baseline?)

**Pull request merged**